services/src/main/java/org/keycloak/protocol/oidc/grants/ciba/endpoints/BackchannelAuthenticationCallbackEndpoint.java

```diff
@@ -81,15 +81,21 @@ public Response processAuthenticationChannelResult(AuthenticationChannelResponse
                     Response.Status.BAD_REQUEST);
         }
 
+        status = preApprove(response);
```

Comment:
Instead of wrapping this here, I'd suggest only to change the line above: 

```
Status status = extractStatusFromResponse(response);
```

... and then someone can override the one method `extractStatusFromResponse`.

Or thinking one step further, add the status to the `BackchannelAuthCallbackContext`, and it can be part of the extendable `verifyAuthenticationRequest()`. If doing that, the rest of the code can be left as is.

Reply:
For our use case this unfortunately doesn't make sense. We don't verify the request itself, we have to check other things (from the database), e.g. our own timeout and other factors which will turn the status to `UNAUTHORIZED` if they are not met.

Reply:
Seems legit. The method `extractStatusFromResponse` should extract some kind of status from the response. But Keycloak might have a saying as well, without checking on any status from the response.